src/thorlabs_cube/driver/kcube/ksc.py

```diff
@@ -0,0 +1,277 @@
+        :param js_max_vel: Maximum velocity for the joystick.
+        :param js_accn: Acceleration for the joystick.
+        :param dir_sense: Direction sense (normal or reversed).
+        :param preset_pos1: Preset position 1.
+        :param preset_pos2: Preset position 2.
+        :param preset_pos3: Preset position 3.
+        :param disp_brightness: Display brightness.
+        :param disp_timeout: Display timeout in ms.
+        :param disp_dim_level: Display dim level.
+        :param js_sensitivity: Joystick sensitivity.
+        """
+
+        # Pack the Channel ID and all the other parameters
+        payload = st.pack(
+            "<HHLLLHHLLLLH",
+            _CHANNEL,
+            js_mode,
+            js_max_vel,
+            js_accn,
+            dir_sense,
+            preset_pos1,
+            preset_pos2,
+            preset_pos3,
+            disp_brightness,
+            disp_timeout,
+            disp_dim_level,
+            js_sensitivity,
+        )
+
+        # Send the message with the packed payload for MOT_SET_KCUBEMMIPARAMS (0x0520)
+        await self.send(Message(MGMSG.MOT_SET_KCUBEMMIPARAMS, data=payload))
+
+    async def get_kcubemmi_params(self):
+        """Get the KCube MMI (joystick) parameters.
+
+        :return: A tuple containing the joystick mode, max velocity,
+        acceleration, direction sense, preset positions, display
+        settings, and joystick sensitivity.
+        """
+
+        # Pack the Channel ID for the request payload
+        payload = st.pack("<H", _CHANNEL)
+
+        # Send the request to get the current joystick parameters
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_KCUBEMMIPARAMS, [MGMSG.MOT_GET_KCUBEMMIPARAMS], data=payload
+        )
+
+        return st.unpack("<HLLLHHLLLLH", get_msg.data[2:])
+
+    async def set_kcubetrigio_config(
+        self, trig1_mode, trig1_polarity, trig2_mode, trig2_polarity
+    ):
+        """Set the KCube trigger I/O configuration parameters.
+
+        :param trig1_mode: Mode for Trigger 1 (input/output).
+        :param trig1_polarity: Polarity for Trigger 1 (high/low).
+        :param trig2_mode: Mode for Trigger 2 (input/output).
+        :param trig2_polarity: Polarity for Trigger 2 (high/low).
+        """
+        # Pack the Channel ID and trigger I/O parameters
+        payload = st.pack(
+            "<HBBBB",
+            _CHANNEL,  # Channel ID is hardcoded to 0x01
+            trig1_mode,
+            trig1_polarity,
+            trig2_mode,
+            trig2_polarity,
+        )
+
+        # Send the message with the packed payload for MOT_SET_KCUBETRIGIOCONFIG (0x0523)
+        await self.send(Message(MGMSG.MOT_SET_KCUBETRIGIOCONFIG, data=payload))
+
+    async def get_kcubetrigio_config(self):
+        """Get the KCube trigger I/O configuration parameters.
+
+        :return: A tuple containing (trig1_mode, trig1_polarity, trig2_mode, trig2_polarity).
```

Comment:
The docstring should tell you what information is returned, not the names of variables.